### DISCUSSION TITLE
Error fetching data from an API

### DISCUSSION THREAD

**Discussion by ChrisB007** [vercel/next.js#29085] [Help] [answered]
Hello guys, newbie here. Can you tell why this code is returning 'undefined'? I was getting undefined while trying to fetch from an API, and now getting the same error trying to log this to the console. This is a NextJS project by the way:

`export default function DataContent({ readText }) {
  console.log(readText); //Returns undefined :(
  return (
    <>
      <ul>
        <li>{/* Map through the project data */}</li>
      </ul>
    </>
  );
}

export async function getServerSideProps() {
  const readText = console.log("This is a test");

  return {
    props: {
      readText
    }
  };
}`

**Comment by wilsonneto-swe** [comment]
Hey man, try this way:

```
export default function DataContent({ readText }) {
  console.log(readText);
  return <>{ readText }</>;
}

export async function getServerSideProps() {
  const readText = "This is a test";
  return {
    props: {
      readText,
    },
  };
}

```

I hope it helps your!

**Reply by ChrisB007** [reply]
@wilsonneto-dev - Thanks for helping. Although I still am getting 'undefined' for some reason. I will continue to dig for solution, but yeah ... thanks for your response.

**Comment by wilsonneto-swe** [comment, marked as answer]
This is weird man, here this code runs well.

I just published the exactly same code here:  https://nextjs-staticprops-simple.vercel.app/
And as you can see, the props is passed and printed on screen, this code is correct, please clear you cache, or send us the error logs.

The deployed repository to check the code: https://github.com/wilsonneto-dev/nextjs-staticprops-simple

![image](https://user-images.githubusercontent.com/20674439/133451485-0308b9e8-823a-4151-acbd-8f91ed20c000.png)

![image](https://user-images.githubusercontent.com/20674439/133451550-10b22605-3441-4efb-972e-2f04e6481450.png)


Let me know if you still getting problems.

I hope it helps you, success on your project.
And check this as the correct answer if it helps you.
Best regards.

**Reply by ChrisB007** [reply]
Thank you so much
